api/src/main/java/com/example/api/service/activity/result/AdditionalPointsService.java

```diff
@@ -0,0 +1,73 @@
+package com.example.api.service.activity.result;
+
+import com.example.api.dto.request.activity.result.AddAdditionalPointsForm;
+import com.example.api.dto.response.activity.task.result.AdditionalPointsResponse;
+import com.example.api.error.exception.WrongUserTypeException;
+import com.example.api.model.activity.result.AdditionalPoints;
+import com.example.api.model.user.User;
+import com.example.api.repo.activity.result.AdditionalPointsRepo;
+import com.example.api.repo.user.UserRepo;
+import com.example.api.security.AuthenticationService;
+import com.example.api.service.validator.UserValidator;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+@Transactional
+public class AdditionalPointsService {
+    private final AdditionalPointsRepo additionalPointsRepo;
+    private final UserRepo userRepo;
+    private final AuthenticationService authService;
+    private final UserValidator userValidator;
+
+    public void saveAdditionalPoints(AddAdditionalPointsForm form) throws WrongUserTypeException {
+        log.info("Saving additional points for student with id {}", form.getStudentId());
+        User user = userRepo.findUserById(form.getStudentId());
+        userValidator.validateStudentAccount(user, form.getStudentId());
+        String professorEmail = authService.getAuthentication().getName();
+        AdditionalPoints additionalPoints;
+        if (form.getDescription() != null) {
+            log.debug("Description is not null");
+            additionalPoints = new AdditionalPoints(null,
+                    user,
+                    form.getPoints(),
+                    form.getDateInMillis(),
+                    professorEmail,
+                    form.getDescription());
```

Comment:
Skoro oba obiekty tworzone w if i else różnią się tylko ostatnim form.getDescription() to nie moglibyśmy wyciągnąć tworzenia tego obiektu tak żeby nie powtarzać całej inicjalizacji ? Jakaś np metoda pomocnicza do której podajemy description lub pusty string i zwraca ten stworzony obiekt.